jsbits/diff.js

```diff
@@ -198,168 +205,152 @@ function syncChildren (os, ns, parent) {
 	    tmp = oldLastIndex - oldFirstIndex;
 	    while (tmp >= 0) {
 	      parent.removeChild(os[oldFirstIndex].domRef);
```

Comment:
I might be missing something here but I think this loop should be incrementing `oldFirstIndex`. Otherwise you keep removing the same element.

Reply:
Nice catch, I removed mutating `os` since this is a base case, but did not realize that `oldFirstIndex` wasn't getting incremented. Thanks. Will fix.